var/spack/repos/builtin/packages/faiss/package.py

```diff
@@ -0,0 +1,131 @@
+    maintainers = ['bhatiaharsh']
+
+    version('1.6.3', sha256='e1a41c159f0b896975fbb133e0240a233af5c9286c09a28fde6aefff5336e542')
+    version('1.5.3', sha256='b24d347b0285d01c2ed663ccc7596cd0ea95071f3dd5ebb573ccfc28f15f043b')
+
+    variant('python', default=False, description='Build Python bindings')
+    variant('tests',  default=False, description='Build Tests')
+
+    conflicts('+tests', when='~python', msg='+tests must be accompanied by +python')
+
+    # we don't want "extend" because we don't want to symlink to python prefix
+    depends_on('python@3.7:',   when='+python', type=('build', 'run'))
+    depends_on('py-numpy',      when='+python', type=('build', 'run'))
+    depends_on('py-scipy',      when='+tests',  type=('build', 'run'))
+
+    depends_on('blas')
+    depends_on('py-setuptools', when='+python', type='build')
+    depends_on('swig',          when='+python', type='build')
+
+    # patch for v1.5.3
+    # faiss assumes that the "source directory" will always
+    # be called "faiss" (not spack-src or faiss-1.5.3)
+    # so, we will have to create a symlink to self (faiss did that in 1.6.3)
+    # and add an include path
+    patch('fixes-in-v1.5.3.patch', when='@1.5.3')
+
+    # patch for v1.6.3
+    # for v1.6.3, GPU build has a bug (two files need to be deleted)
+    # https://github.com/facebookresearch/faiss/issues/1159
+    # also, some include paths in gpu/tests/Makefile are missing
+    patch('fixes-in-v1.6.3.patch', when='@1.6.3')
+
+    def configure_args(self):
+        args = []
+        args.extend(self.with_or_without('cuda', activation_value='prefix'))
+        return args
+
+    def build(self, spec, prefix):
+
+        make()
+
+        if '+python' in self.spec:
+            make('-C', 'python')
+
+        # CPU tests
+        if '+tests' in self.spec:
+            with working_dir('tests'):
+                make('gtest')
+                make('tests')
+
+        # GPU tests
+        if '+tests' in self.spec and '+cuda' in self.spec:
+            with working_dir(os.path.join('gpu', 'test')):
+                make('gtest')
+                make('build')                       # target added by the patch
+                make('demo_ivfpq_indexing_gpu')
+
+    def install(self, spec, prefix):
+
+        make('install')
+
+        if '+python' in self.spec:
+            with working_dir('python'):
+                setup_py('install', '--prefix=' + prefix,
+                         '--single-version-externally-managed', '--root=/')
+
+        if not os.path.isdir(self.prefix.bin):
+            os.makedirs(self.prefix.bin)
+
+        def _prefix_and_install(file):
+            os.system('mv {} faiss_{}'.format(file, file))
+            install('faiss_{}'.format(file), self.prefix.bin)
+
+        # CPU tests
+        if '+tests' in self.spec:
+            with working_dir('tests'):
+                # rename the exec to keep consistent with gpu tests
+                os.system('mv tests TestCpu')
```

Comment:
Use a builtin Python utility here. We hope to add Windows support in the near future.